internal/services/servicebus/servicebus_namespace_resource.go

```diff
@@ -336,6 +344,16 @@ func resourceServiceBusNamespaceCreateUpdate(d *pluginsdk.ResourceData, meta int
 		parameters.Sku.Capacity = utils.Int64(int64(capacity.(int)))
 	}
 
+	if premiumMessagingUnit := d.Get("premium_messaging_partitions"); premiumMessagingUnit != nil {
+		if !strings.EqualFold(sku, string(namespaces.SkuNamePremium)) && premiumMessagingUnit.(int) > 0 {
+			return fmt.Errorf("Premium messaging partition is not supported by service Bus SKU %q and it can only be set to 0", sku)
+		}
+		if strings.EqualFold(sku, string(namespaces.SkuNamePremium)) && premiumMessagingUnit.(int) == 0 {
+			return fmt.Errorf("Service Bus SKU %q only supports `premium_messaging_partitions` of 1, 2, 4", sku)
+		}
```

Comment:
This doesn't match the schema validation above? The property defaults to `0` also.

Reply:
This is for the premium namespace, 0 is not a valid value for premium namespace.